src/main/java/com/autotune/experimentManager/handler/util/EMStatusUpdateHandler.java

```diff
@@ -0,0 +1,104 @@
+                }
+            });
+        });
+        int iterationCount = workflow.size() ;
+        int stepsCountInt = stepsCount.intValue();
+        int total = iterationCount * stepsCountInt ;
+        if (queuedCount.intValue() == total){
+            cycleMetaData.setBeginTimestamp(new Timestamp(System.currentTimeMillis()));
+            cycleMetaData.setStatus(EMUtil.EMExpStatus.IN_PROGRESS);
+        }else if(completedOrFailedCount.intValue() == total){
+            cycleMetaData.setEndTimestamp(new Timestamp(System.currentTimeMillis()));
+            cycleMetaData.setStatus(EMUtil.EMExpStatus.COMPLETED);
+        }else {
+            LOGGER.debug("Cycle Iterations still in progress {} / {} ", completedOrFailedCount.intValue() , total);
+        }
+    }
+
+    public static void updateTrialMetaDataStatus(ExperimentTrial experimentTrial,TrialDetails trialDetails) {
+        AtomicInteger queuedCount = new AtomicInteger(0);
+        AtomicInteger completedOrFailedCount = new AtomicInteger(0);
+        trialDetails.getTrialMetaData().getCycles().forEach((cycleName,cycleMetaData) -> {
+            if(cycleMetaData.getStatus().equals(EMUtil.EMExpStatus.QUEUED)){
+                queuedCount.set(queuedCount.get() + 1);
+            }else if (cycleMetaData.getStatus().equals(EMUtil.EMExpStatus.COMPLETED)){
+                completedOrFailedCount.set(completedOrFailedCount.get() + 1);
+            }else if (cycleMetaData.getStatus().equals(EMUtil.EMExpStatus.FAILED)){
+                completedOrFailedCount.set(completedOrFailedCount.get() + 1);
+            }
+        });
+        trialDetails.getTrialMetaData().getTrialWorkflow().forEach((stepName,stepsMetaData) -> {
+            if(stepsMetaData.getStatus().equals(EMUtil.EMExpStatus.QUEUED)) queuedCount.set(queuedCount.get() + 1);
+            else if(stepsMetaData.getStatus().equals(EMUtil.EMExpStatus.COMPLETED)) completedOrFailedCount.set(completedOrFailedCount.get() + 1);
+            else if(stepsMetaData.getStatus().equals(EMUtil.EMExpStatus.FAILED)) completedOrFailedCount.set(completedOrFailedCount.get() + 1);
+        });
+        int totalSteps = trialDetails.getTrialMetaData().getCycles().size() + trialDetails.getTrialMetaData().getTrialWorkflow().size() ;
+        if(queuedCount.intValue() == totalSteps){
+            trialDetails.getTrialMetaData().setBeginTimestamp(new Timestamp(System.currentTimeMillis()));
+            trialDetails.getTrialMetaData().setStatus(EMUtil.EMExpStatus.IN_PROGRESS);
+        }
+        else if(completedOrFailedCount.intValue() == totalSteps){
+            trialDetails.getTrialMetaData().setEndTimestamp(new Timestamp(System.currentTimeMillis()));
+            trialDetails.getTrialMetaData().setStatus(EMUtil.EMExpStatus.COMPLETED);
+        }else {
+            LOGGER.debug("Cycle still in progress {} / {} ", completedOrFailedCount.intValue() , trialDetails.getTrialMetaData().getCycles().size());
+        }
```

Comment:
spacing consistency and also use brackets for the statements inside `if` and `else`